src/main/java/com/userservice/Repository/UserRepository.java

```diff
@@ -0,0 +1,16 @@
+package com.userservice.Repository;
+
+
+import com.userservice.Entity.Users;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.UUID;
+
+
+@Repository
+public interface UserRepository extends JpaRepository<Users, UUID> {
+    Users findByUsername(String name);
```

Comment:
`findByUsername` will not be required as we are not maintaining the uniqueness of the `username`, instead we can use `Users findByPhoneNumber(CountryCode code, Long phoneNumber)`. 
In our application, users can have the same username.